packages/core/src/domain/conversion/bundle-modifications/__tests__/post-process-bundle.test.ts

```diff
@@ -0,0 +1,225 @@
+      condition.extension = [documentExtension];
+      const bundle = makeBundle({ entries: [condition], type: "collection" });
+
+      const updatedBundle = bundleMods.replaceIdsForResourcesWithDocExtension(bundle, patientId);
+      const updCondition = updatedBundle.entry?.[0]?.resource;
+      expect(updCondition).toBeDefined();
+      expect(updCondition?.id).toBeDefined();
+      expect(updCondition?.id).not.toBe(condition.id);
+      expect(updCondition?.meta?.source).toBe(fileName);
+      expect(updCondition).toMatchObject({
+        ...condition,
+        id: updCondition?.id,
+        meta: { source: fileName },
+      });
+    });
+
+    it("throws error when bundle entry is empty", () => {
+      const { patientId } = initTest();
+      const bundle = makeBundle({ entries: [], type: "collection" });
+      expect(() => bundleMods.replaceIdsForResourcesWithDocExtension(bundle, patientId)).toThrow(
+        "Missing bundle entries"
+      );
+    });
+
+    it("throws error when bundle has no entries", () => {
+      const { patientId } = initTest();
+      const bundle = makeBundle({ type: "collection" });
+      expect(() => bundleMods.replaceIdsForResourcesWithDocExtension(bundle, patientId)).toThrow(
+        "Missing bundle entries"
+      );
+    });
+  });
+
+  describe("addExtensionToConversion", () => {
+    it("adds a document extension to a resource that did not have it", () => {
+      const { condition, documentExtension } = initTest();
+      const bundle = makeBundle({ entries: [condition], type: "collection" });
+
+      const updatedBundle = bundleMods.addExtensionToConversion(bundle, documentExtension);
+      const updCondition = updatedBundle.entry?.[0]?.resource as Condition | undefined;
+      expect(updCondition).toBeDefined();
+      if (!updCondition) throw new Error("Bundle entry got removed!");
+
+      expect(updCondition.extension).toHaveLength(1);
+      expect(updCondition.extension?.[0]).toEqual(documentExtension);
+      expect(condition.extension).toBeUndefined();
+    });
+
+    it("appends the document extension to a resource with existing extensions", () => {
+      const { condition, documentExtension } = initTest();
+
+      const existingExtension = {
+        url: "http://example.com/existing",
+        valueString: "test",
+      };
+      condition.extension = [existingExtension];
+      const bundle = makeBundle({ entries: [condition], type: "collection" });
+
+      const updatedBundle = bundleMods.addExtensionToConversion(bundle, documentExtension);
+      const updCondition = updatedBundle.entry?.[0]?.resource as Condition | undefined;
+      expect(updCondition).toBeDefined();
+      if (!updCondition) throw new Error("Bundle entry got removed!");
+
+      expect(updCondition.extension).toHaveLength(2);
+      expect(updCondition.extension).toEqual(
+        expect.arrayContaining([existingExtension, documentExtension])
+      );
+      expect(condition.extension).toHaveLength(1);
```

Comment:
ditto: prob move to before the call to the main function being tested, `addExtensionToConversion`